src/cli/cmd/init/mod.rs

```diff
@@ -0,0 +1,347 @@
+    #[clap(long, short, default_value = ".")]
+    root: PathBuf,
+
+    #[clap(long, short, default_value = "./flake.nix")]
+    output: PathBuf,
+
+    #[clap(from_global)]
+    api_addr: url::Url,
+}
+
+#[async_trait::async_trait]
+impl CommandExecute for InitSubcommand {
+    async fn execute(self) -> Result<ExitCode> {
+        let mut inputs: HashMap<String, String> = HashMap::new();
+        let mut dev_shell_packages: Vec<String> = Vec::new();
+        let mut dev_shells: HashMap<String, DevShell> = HashMap::new();
+        let mut overlay_refs: Vec<String> = Vec::new();
+        let mut overlay_attrs: HashMap<String, String> = HashMap::new();
+        let mut env_vars: HashMap<String, String> = HashMap::new();
+
+        if self.output.exists() && !Prompt::bool("A flake.nix already exists in the current directory. Would you like to overwrite it?")? {
+            println!("Exiting. Let's a build a new flake soon, though :)");
+            return Ok(ExitCode::SUCCESS);
+        }
+
+        println!("Let's build a Nix flake!");
+
+        let project = Project::new(self.root);
+        let description = Prompt::maybe_string("An optional description for your flake")?;
+
+        fn get_systems() -> Result<Vec<String>, FhError> {
+            let selected = Prompt::guided_multi_select(
+                "Which systems would you like to support?",
+                "system",
+                SYSTEMS.to_vec(),
+            )?;
+
+            if selected.is_empty() {
+                println!("❌ You need to select at least one system to support");
+                #[allow(clippy::needless_return)]
+                return get_systems();
+            } else {
+                Ok(selected)
+            }
+        }
+
+        let systems = get_systems()?;
+
+        // We could conceivably create a version of `fh init` Nixpkgs included only if certain other choices
+        // are made. But for the time being so much relies on it that we don't have a great opt-out story,
+        // so best to just include it in all flakes.
+        let nixpkgs = match Prompt::select(
+            "Which Nixpkgs version would you like to include?",
+            &[
+                "23.05",
+                "latest",
+                "unstable",
+                "select a specific release (not recommended in most cases)",
+            ],
+        )?
+        .as_str()
+        {
+            // MAYBE: find an enum-based approach to this
+            "23.05" => String::from("0.2305.*"),
+            "latest" => String::from("*"),
+            "unstable" => String::from("0.1.*"),
+            "select a specific release (not recommended in most cases)" => {
+                select_nixpkgs(&self.api_addr).await?
+            }
+            _ => String::from("*"), // Unreachable
+        };
+
+        inputs.insert(
+            String::from("nixpkgs"),
+            format!("https://flakehub.com/f/NixOS/nixpkgs/{nixpkgs}.tar.gz"),
+        );
+
+        // Go projects
```

Comment:
I think it would be much more maintainable if we were to split these out into their own (sub)modules. I could imagine something like this:

```rust
Golang::init(&mut inputs, &mut dev_shell_packages);
// where `Golang::init` is defined in src/cli/cmd/init/langs/golang.rs
```

and the same for all the other languages.